crates/oxc_linter_plugin/src/plugin.rs

```diff
@@ -0,0 +1,615 @@
+}
+
+#[derive(Deserialize, Clone, Debug)]
+pub struct InputQuery {
+    pub name: String,
+    pub query: String,
+    pub args: BTreeMap<Arc<str>, TransparentValue>,
+    pub summary: String,
+    pub reason: String,
+    #[serde(skip_deserializing)]
+    pub path: PathBuf,
+    pub post_transform: Option<String>,
+    #[serde(default)]
+    pub tests: QueryTests,
+}
+
+#[derive(Deserialize, Default, Clone, Debug)]
+pub struct QueryTests {
+    pub pass: Vec<SingleTest>,
+    pub fail: Vec<SingleTest>,
+}
+
+#[derive(Deserialize, Clone, Debug)]
+pub struct SingleTest {
+    pub relative_path: Vec<String>,
+    pub code: String,
+}
+
+#[derive(Debug)]
+pub struct LinterPlugin {
+    pub(super) rules: Vec<InputQuery>,
+    schema: &'static Schema,
+}
+
+pub enum RulesToRun {
+    All,
+    Only(String),
+}
+
+#[derive(Debug, Error, Diagnostic)]
+#[error(transparent)]
+pub struct ParseError(serde_yaml::Error);
+
+#[derive(Debug, Error, Diagnostic)]
+pub enum ErrorFromLinterPlugin {
+    #[error("{0}")]
+    PluginGenerated(String, String, #[label("{1}")] Span),
+    #[error("{error_message}")]
+    Trustfall {
+        error_message: String,
+        #[source_code]
+        query_source: Arc<NamedSource>,
+        #[label = "This query failed."]
+        query_span: SourceSpan,
+    },
+    #[error(transparent)]
+    Ignore(ignore::Error),
+    #[error(transparent)]
+    ReadFile(std::io::Error),
+    #[error("Failed to parse file at path: {0}")]
+    QueryParse(PathBuf, #[related] Vec<ParseError>),
+    #[error(
+        "Expected span_start and span_end to be List of Int or Int, instead got\nspan_start = {span_start}\nspan_end = {span_end}"
+    )]
+    WrongTypeForSpanStartSpanEnd {
+        span_start: String,
+        span_end: String,
+        #[source_code]
+        query_source: Arc<NamedSource>,
+        #[label = "This query failed."]
+        query_span: SourceSpan,
+    },
+}
+
+impl LinterPlugin {
+    pub fn new(schema: &'static Schema, queries_path: &PathBuf) -> oxc_diagnostics::Result<Self> {
+        let mut deserialized_queries = vec![];
+
+        for dir_entry_found_maybe in Walk::new(queries_path) {
+            let dir_entry_found = dir_entry_found_maybe.map_err(ErrorFromLinterPlugin::Ignore)?;
+
+            let pathbuf = dir_entry_found.path().to_path_buf();
+
+            if pathbuf.extension().map_or(false, |ext| ext.eq_ignore_ascii_case("yml")) {
+                let text = fs::read_to_string(&pathbuf).map_err(ErrorFromLinterPlugin::ReadFile)?;
+
+                let mut deserialized =
+                    serde_yaml::from_str::<InputQuery>(&text).map_err(|err| {
+                        ErrorFromLinterPlugin::QueryParse(pathbuf.clone(), vec![ParseError(err)])
+                    })?;
+                deserialized.path = pathbuf;
+                deserialized_queries.push(deserialized);
+            }
+        }
+
+        Ok(Self { rules: deserialized_queries, schema })
+    }
+
+    #[allow(clippy::too_many_lines)]
```

Comment:
I think clippy is onto something here. Try to break up this function into smaller, more manageable parts.